src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,201 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+    private static class Node<T>{
+        T item;
+        Node<T> prev;
+        Node<T> next;
+
+        Node(Node<T> prev, T item, Node<T> next){
+            this.prev = prev;
+            this.next = next;
+            this.item = item;
+        }
+    }
+
+    // Adds new value at the last position
     @Override
     public void add(T value) {
+        if (tail != null) {
+            tail.next = new Node<>(tail, value, null);
+            tail = tail.next;
+        } else {
+            head = tail = new Node<>(null, value, null);
+        }
+        size++;
     }
 
+    // Adds new value at the index position or if index is out of size,
+    // it throws IndexOutOfBoundsException
     @Override
     public void add(T value, int index) {
+        if (IndexOutOfBoundsException(index)) {
```

Comment:
Please replace ```if (IndexOutOfBoundsException(index)) {``` by a check, which will compare ```index``` and ```size```

Reply:
Thanks for your suggestion! Done.